### PR TITLE
Test if microversion value not nil and not empty

### PR DESCRIPTION
**Pull request by shoudusse**
This PR fixes https://github.com/fog/fog-openstack/issues/359

### CODE REVIEW

File: lib/fog/openstack/core.rb [hunk 1]
```diff
@@ -145,7 +145,7 @@ def microversion_newer_than?(version)
 
       def headers(additional_headers)
         additional_headers ||= {}
-        if @microversion
+        unless @microversion.nil? or @microversion.empty?
```
Comment (gildub) [inline lib/fog/openstack/core.rb]: Please use `&&` operator instead of `or`.

**Review by gildub — CHANGES_REQUESTED** [review]

**Comment by gildub** [comment]
@shoudusse, LGTM besides the operator change. Thanks

File: lib/fog/openstack/core.rb (hunk 1 above)
Comment (shoudusse) [reply lib/fog/openstack/core.rb]: @gildub,  Don't you mean `||` instead of `or`?

Comment (gildub) [reply lib/fog/openstack/core.rb]: Yes indeed! `or` as well as `and` are not recommended.

**Comment by gildub** [comment]
@shoudusse, Please do not merge your own code. I know it's tempting and we have all been there (including me!).

Although sometimes there are exceptions such as emergencies (broken CI, etc) and cases where it doesn't impact the code directly such as documentation or tests.
Also we don't enforce 2 peers review, it's a good practice especially about complex issues,  
I believe the practice is to always have at least someone else merge your patch to guarantee at least 1 peer review.

Thanks

**Pull request merged**